content/en/docs/v3.6/learning/api_guarantees.md

```diff
@@ -106,9 +109,35 @@ expired by the wall clock time to live (TTL). However, users need to be aware
 about [the important properties of the APIs and usage][why] for implementing
 correct distributed coordination mechanisms.
 
+## etcd specific definitions
+
+### Operation completed
+
+An etcd operation is considered complete when it is committed through consensus,
+and therefore “executed” -- permanently stored -- by the etcd storage engine.
+The client knows an operation is completed when it receives a response from the
+etcd server. Note that the client may be uncertain about the status of an
+operation if it times out, or there is a network disruption between the client
+and the etcd member. etcd may also abort operations when there is a leader
+election. etcd does not send `abort` responses to  clients’ outstanding requests
+in this event.
+
+### Revision
+
+An etcd operation that modifies the key value store is assigned a single
+increasing revision. A transaction operation might modify the key value store
+multiple times, but only one revision is assigned. The revision attribute of a
+key value pair that was modified by the operation has the same value as the
+revision of the operation. The revision can be used as a logical clock for key
+value store. A key value pair that has a larger revision is modified after a key
+value pair with a smaller revision. Two key value pairs that have the same
+revision are modified by an operation "concurrently".
```

Comment:
Because important part should be at the top. You cannot ask someone reading the document about api guarantees to go first for very wordy section of definitions first. You give them the important parts first and if they don't know some term, they can use link to jump to their definition.

Reply:
I can also list counter-reasons, but it isn't a big deal, so I won't insist on it.